code/modules/clothing/masks/gasmask.dm

```diff
@@ -45,15 +54,65 @@ GLOBAL_LIST_INIT(clown_mask_options, list(
 	QDEL_LAZYLIST(gas_filters)
 	return..()
 
+/obj/item/clothing/mask/gas/equipped(mob/equipee, slot)
+	if(cig)
+		cig.equipped(equipee, slot)
+	return ..()
+
+/obj/item/clothing/mask/gas/adjustmask(mob/living/carbon/user)
+	if(isnull(cig))
+		return ..()
+	balloon_alert(user, "there's a cig in the way!")
+
+
 /obj/item/clothing/mask/gas/examine(mob/user)
 	. = ..()
-	if(max_filters > 0)
-		. += "<span class='notice'>[src] has [max_filters] slot\s for filters.</span>"
+	if(cig)
+		. += span_notice("There is a [cig.name] jammed into the filter slot.")
+	if(max_filters > 0 && !cig)
+		. += span_notice("[src] has [max_filters] slot\s for filters.")
 	if(LAZYLEN(gas_filters) > 0)
-		. += "<span class='notice'>Currently there [LAZYLEN(gas_filters) == 1 ? "is" : "are"] [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.</span>"
-		. += "<span class='notice'>The filters can be removed by right-clicking with an empty hand on [src].</span>"
+		. += span_notice("Currently there [LAZYLEN(gas_filters) == 1 ? "is" : "are"] [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.")
+		. += span_notice("The filters can be removed by right-clicking with an empty hand on [src].")
+
+/obj/item/clothing/mask/gas/proc/on_cig_destroyed()
+	SIGNAL_HANDLER
+	if(cig)
+		cig = null
+		if(istype(src.loc, /mob))
+			var/mob/wearer = src.loc
+			wearer.update_worn_mask()
 
 /obj/item/clothing/mask/gas/attackby(obj/item/tool, mob/user)
+	if(istype(tool, /obj/item/clothing/mask/cigarette))
+		var/valid_wearer = ismob(src.loc)
+
+		if(flags_cover & MASKCOVERSMOUTH)
+			balloon_alert(user, "mask's mouth is covered!")
+			return ..()
+
+		if(max_filters <= 0 || cig)
+			balloon_alert(user, "can't hold filters!")
+			return ..()
+
+		if(has_filter)
+			balloon_alert(user, "filters in the mask!")
+			return ..()
+
+		cig = tool
+		if(valid_wearer)
+			cig.equipped(src.loc, ITEM_SLOT_MASK)
```

Comment:
Note that this is misleading - just because the loc is a mob doesn't mean it's equipped to mask. it might be in hands. 

You may want to use `get_slot_by_item` and pass the slot it returns to equipped instead.